models/us_treasury_usaspending/code/dictionary_coverage.py

```
@@ -0,0 +1,95 @@
+    cols = dictionary_columns(table)
+    if not cols:
+        return {"covered": [], "gaps": {}}
+    ref = f"`{project}.{DATASET}.{table}`"
+    dic = f"`{project}.{DATASET}.dicionario`"
+    unions = "\nunion all\n".join(
+        f"""select '{c}' as column_name, t.{c} as value, count(*) as n
+from {ref} t
+left join (select chave from {dic} where nome_coluna = '{c}' and id_tabela = '{table}') d
+  on d.chave = t.{c}
+where t.{c} is not null and d.chave is null
+group by 1, 2"""
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)